nsxt/resource_nsxt_lb_service_test.go

```diff
@@ -0,0 +1,204 @@
+		if !ok {
+			return fmt.Errorf("NSX load balancer service resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("NSX load balancer service resource ID not set in resources ")
+		}
+
+		service, responseCode, err := nsxClient.ServicesApi.ReadLoadBalancerService(nsxClient.Context, resourceID)
+		if err != nil {
+			return fmt.Errorf("Error while retrieving load balancer service ID %s. Error: %v", resourceID, err)
+		}
+
+		if responseCode.StatusCode != http.StatusOK {
+			return fmt.Errorf("Error while checking if load balancer service %s exists. HTTP return code was %d", resourceID, responseCode.StatusCode)
+		}
+
+		if displayName == service.DisplayName {
+			return nil
+		}
+		return fmt.Errorf("NSX load balancer service %s wasn't found", displayName)
+	}
+}
+
+func testAccNSXLbServiceCheckDestroy(state *terraform.State, displayName string) error {
+	nsxClient := testAccProvider.Meta().(*nsxt.APIClient)
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_lb_service" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		service, responseCode, err := nsxClient.ServicesApi.ReadLoadBalancerService(nsxClient.Context, resourceID)
+		if err != nil {
+			if responseCode.StatusCode != http.StatusOK {
+				return nil
+			}
+			return fmt.Errorf("Error while retrieving load balancer service ID %s. Error: %v", resourceID, err)
+		}
+
+		if displayName == service.DisplayName {
+			return fmt.Errorf("NSX load balancer service %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNSXLbCreateTopology() string {
+	edgeClusterName := getEdgeClusterName()
+	tier0Name := getTier0RouterName()
+	return fmt.Sprintf(`
+data "nsxt_edge_cluster" "EC" {
+  display_name = "%s"
+}
+
+data "nsxt_logical_tier0_router" "test" {
+  display_name = "%s"
+}
+
+resource "nsxt_logical_router_link_port_on_tier0" "test" {
+  display_name      = "port_on_tier0"
+  logical_router_id = "${data.nsxt_logical_tier0_router.test.id}"
+}
+
+resource "nsxt_logical_tier1_router" "test" {
+  display_name    = "test"
+  edge_cluster_id = "${data.nsxt_edge_cluster.EC.id}"
+}
+
+resource "nsxt_logical_router_link_port_on_tier1" "test" {
+  logical_router_id             = "${nsxt_logical_tier1_router.test.id}"
+  linked_logical_router_port_id = "${nsxt_logical_router_link_port_on_tier0.test.id}"
+}`, edgeClusterName, tier0Name)
+}
+
+func testAccNSXLbServiceCreateTemplate() string {
+	return testAccNSXLbCreateTopology() + `
+resource "nsxt_lb_service" "test" {
+  display_name      = "test"
+  enabled           = true
+  description       = "Acceptance Test"
+  logical_router_id = "${nsxt_logical_tier1_router.test.id}"
+  size              = "SMALL"
+  error_log_level   = "DEBUG"
+
+  tag {
+    scope = "scope1"
+    tag   = "tag1"
+  }
+
+  depends_on        = ["nsxt_logical_router_link_port_on_tier1.test"]
```

Comment:
this line should be above the tag

Reply:
Is this a must? I wanted to emphasize depends_on, hence the location and spacing

Reply:
You can put this configuration in a tf file and run terraform fmt main.tf to see if it was re-formatted.
Seems ok when I tried it, just that the extra spaces in line 178 should be removed.